.github/workflows/reusable-module-prep-release.yml

```diff
@@ -81,53 +101,131 @@ jobs:
         run: composer validate
 
       - name: Install PHP Dependencies
-        run: composer install --no-dev --optimize-autoloader
```

Comment:
Normally I don't like to advocate for adding a 3rd-party dependency. But in this case, we should utilize the https://github.com/ramsey/composer-install/. It handles a ton of Composer related things for us behind the scenes including caching (which would eliminate two of the previous steps).

Reply:
Nice, I've got that added and was able to eliminated the cache related steps like you pointed out.